src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs

```diff
@@ -69,7 +69,7 @@ private void VisitArrayType(IArrayTypeSymbol symbol, TypeWithAnnotations typeOpt
             }
 
             var arrayType = symbol;
-            while (arrayType != null && arrayType != underlyingType)
+            while (arrayType != null && !Equals(arrayType, underlyingType))
```

Comment:
Does it make sense to use the static `Object.Equals(objA, objB)` here instead of e.g. `TypeSymbol.Equals(otherType, typeCompareKind)`?

Reply:
It uses `global::System.Object.Equals`: https://github.com/dotnet/roslyn-analyzers/blob/3919e6ec530caf90b11058691c1876465bd7f75c/src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/Fixers/CompareSymbolsCorrectlyFix.cs#L54-L56

Reply:
OK, reading more of the context makes it seem like this is fine.